x-pack/plugins/apm/public/components/shared/charts/CustomPlot/StaticPlot.js

```diff
@@ -56,13 +57,14 @@ class StaticPlot extends PureComponent {
       case 'areaMaxHeight':
         const yMax = last(plotValues.yTickValues);
         const data = serie.data.map(p => ({
+          x0: p.x0,
           x: p.x,
           y0: 0,
-          y: p.y ? yMax : null
+          y: yMax
         }));
 
         return (
-          <AreaSeries
+          <VerticalRectSeries
```

Comment:
This line might be the biggest change in this PR: going from AreaSeries to RectSeries. 
An AreaSeries take points like `{ x, y0, y }` - so no horisontal width. To set the horisontal width I used to add a starting point, and ending point, and then padded with a `null ` point to avoid connecting the area with the following area. Pretty hacky:
https://github.com/elastic/kibana/blob/54f1a739cd7c1b259d6105a63abcc4a4a4b4906e/x-pack/plugins/apm/public/store/selectors/chartSelectors.ts#L223-L237

`RectSeries` take points like `{ x0, x, y0, y }`. So the hacky stuff can go away.